app/runner.js

```diff
@@ -4,81 +4,85 @@ const path = require('path');
     }
 
-    if (options.env) {
-        execOptions.env = Object.assign(options.env, process.env);
+    static processPath(dir) {
+        dir = untildify(dir).trim();
+        return path.resolve(dir);
     }
 
-    const arr = command.split(" ").concat(args);
-    let proc;
-    try {
-        proc = childProcess.spawn(arr.shift(), arr, execOptions);
-    } catch (e) {
-        done(1, [], [e]);
-        return null;
-    }
+    static runProcess(command, dir, options, done) {
+        const execOptions = {
+            shell: true
+        };
+        let args = [];
+        if (dir) execOptions.cwd = Runner.processPath(dir);
 
-    let outs = "";
-    let errs = "";
+        if (options.args) {
+            if (options.args.constructor === Array) {
+                args = options.args;
+            } else args = options.args.split(" ");
+        }
 
-    proc.stdout.on('data', (data) => {
-        outs += data;
-        if (options.stdout) options.stdout(data.toString());
-    });
+        if (options.env) {
+            execOptions.env = Object.assign(options.env, process.env);
+        }
 
-    proc.stderr.on('data', (data) => {
-        errs += data;
-        if (options.stderr) options.stderr(data.toString());
-    });
+        const arr = command.split(" ").concat(args);
+        let proc;
+        try {
+            proc = childProcess.spawn(arr.shift(), arr, execOptions);
+        } catch (e) {
+            done(1, [], [e]);
+            return null;
+        }
 
-    proc.on('error', (err) => {
-        errs += err;
-    });
+        let outs = "";
+        let errs = "";
 
-    proc.on('close', (code, signal) => {
-        if (signal === "SIGTERM" && code === null) code = 143;
-        if (done) done(code, filterOutput(outs), filterOutput(errs));
-    });
-    return proc;
-}
+        proc.stdout.on('data', (data) => {
+            outs += data;
+            if (options.stdout) options.stdout(data.toString());
+        });
 
+        proc.stderr.on('data', (data) => {
+            errs += data;
+            if (options.stderr) options.stderr(data.toString());
+        });
 
-module.exports = function(command, dir, options, done) {
-    if (!done && typeof options === 'function') {
-        done = options;
-        options = null;
-    }
-    if (!done && !options && typeof dir === 'function') {
-        done = dir;
-        dir = null;
-        options = null;
-    }
-    if (!options && typeof dir === 'object') {
-        options = dir;
-        dir = null;
+        proc.on('error', (err) => {
+            errs += err;
+        });
+
+        proc.on('close', (code, signal) => {
+            if (signal === "SIGTERM" && code === null) code = 143;
+            if (done) done(code, Runner.filterOutput(outs), Runner.filterOutput(errs));
+        });
+        return proc;
     }
-    if (!options) options = {};
 
-    return runProcess(command, dir, options, done);
-};
+    static run(command, dir, options, done) {
```

Comment:
Yeah OK thats OK, and please don't use `process` as a variable name, this is not related to this comment but i forgot to say you. Node.js users should get errors using `process` as variable name because it's a global!